node/src/reactor/event_queue_metrics.rs

```diff
@@ -0,0 +1,80 @@
+use prometheus::{self, IntGauge, Registry};
+
+use crate::reactor::{EventQueueHandle, QueueKind};
+use itertools::Itertools;
+use std::collections::HashMap;
+use tracing::debug;
+
+#[derive(Debug)]
+pub(super) struct EventQueueMetrics {
+    event_queue_gauges: HashMap<QueueKind, IntGauge>,
+    event_total: IntGauge,
+    registry: Registry,
+}
+
+impl EventQueueMetrics {
+    /// Initializes event queue sizes metrics.
+    pub(super) fn new<REv: 'static>(
+        registry: Registry,
+        event_queue_handle: EventQueueHandle<REv>,
+    ) -> Result<Self, prometheus::Error> {
+        let mut event_queue_gauges: HashMap<QueueKind, IntGauge> = HashMap::new();
+        for queue_kind in event_queue_handle.event_counts().keys() {
+            let key = format!("scheduler_queue_{:?}_count", queue_kind);
+            let queue_event_counter = IntGauge::new(key, "Event in the queue.".to_string())?;
+            registry.register(Box::new(queue_event_counter.clone()))?;
+            let result = event_queue_gauges.insert(*queue_kind, queue_event_counter);
+            assert!(result.is_none(), "Map keys should not be overwritten.");
+        }
+
+        let event_total = IntGauge::new(
+            "scheduler_queue_total_count",
+            "total count of events in queues.",
+        )?;
+        registry.register(Box::new(event_total.clone()))?;
+
+        Ok(EventQueueMetrics {
+            event_queue_gauges,
+            event_total,
+            registry,
+        })
+    }
+
+    pub(super) fn estimate<REv: 'static>(&self, event_queue_handle: &EventQueueHandle<REv>) {
```

Comment:
Aside from the missing docs, maybe we don't need to call this estimate, but `event_counts` or something like that? It's not an estimate, it is fairly accurate, I would say.

Reply:
We should also document possible inaccuracies at this point.

Reply:
What inaccuracies?

Reply:
There a small race condition when querying for queue sizes, while events are counted, another client might be removing an event in the interim. It's not likely to even occur in our system, since our event processing so centralized.

Reply:
Done.